src/URI.js

```diff
@@ -1223,6 +1227,26 @@
   };
 
   // compound accessors
+  p.origin = function(v, build) {
+    var parts;
+
+    if (this._parts.urn) {
+      return v === undefined ? '' : this;
+    }
+
+    if (v === undefined) {
+      return this._parts.hostname ? URI.buildOrigin(this._parts) : '';
```

Comment:
Sure but then you can get the reverse too: `http://` So the whole origin should be empty if there is no authority. 

e.g.
```
URI('http:///foo.html').origin() === ''
```